internal/repositories/sql/user/user.go

```diff
@@ -51,3 +52,43 @@ func (i *impl) GetUser(req *user.GetUserRequest) (*user.GetUserResponse, error)
 		User: model,
 	}, nil
 }
+
+func (i *impl) CheckIfUserExists(req *user.CheckIfUserExistsRequest) (*user.CheckIfUserExistsResponse, error) {
+	userData := &models.User{}
+	exec := i.db.Master().First(userData, "email_id = ?", req.EmailId)
+
+	if exec.Error != nil {
+		i.logger.Error("ERROR_CHECK_IF_USER_EXISTS", zap.Error(exec.Error))
+		return nil, exec.Error
+	}
+
+	if errors.Is(exec.Error, gorm.ErrRecordNotFound) {
+		i.logger.Info("user not found", zap.String("email_id", req.EmailId))
+		return &user.CheckIfUserExistsResponse{
+			IsExisting: false,
+		}, nil
+	}
+
+	return &user.CheckIfUserExistsResponse{
+		IsExisting: true,
+	}, nil
+}
+
+func (i *impl) GetUserByEmailId(req *user.GetUserByEmailIdRequest) (*user.GetUserByEmailIdResponse, error) {
+	userData := &models.User{}
+	exec := i.db.Master().First(userData, "email_id=?", req.EmailId)
+
+	if exec.Error != nil {
+		i.logger.Error("ERROR_GET_USER_BY_EMAIL_ID", zap.Error(exec.Error))
+		return nil, exec.Error
+	}
+
+	if errors.Is(exec.Error, gorm.ErrRecordNotFound) {
+		i.logger.Info("user not found", zap.String("email_id", req.EmailId))
+		return nil, nil
+	}
+
+	return &user.GetUserByEmailIdResponse{
+		User: userData,
+	}, nil
+}
```

Comment:
Should we combine these two functions and check for user in `GetUserByEmailId` function only?